akirachix/Upile-Frontend
# Feature/unidentifiedbodies

**Engineers Checklist**

**Summary of Change**
This PR introduces a new page and form for registering and managing Unidentified Bodies in the Police and Mortuary Dashboard. The feature allows users (police and mortuary staff) to input details about unidentified bodies, including case information, physical characteristics, and location data. This form is critical for maintaining accurate records and improving the search and identification processes for unknown individuals.

**Motivation and Context**
The motivation behind this feature is to provide a structured and organized method for recording unidentified bodies, which is a vital part of the mortuary services. The ability to log this information in a centralized system helps both the police and mortuary staff manage their cases more efficiently, making the identification process quicker and more effective. This new page is part of the ongoing development to streamline handling missing and unidentified persons.

**Dependencies**
No new external dependencies are required. However, the form relies on existing API endpoints to submit data, including patient information, case status, and location data.

**What this PR does**
This PR implements an Unidentified Body Page and Form on the Police and Mortuary Dashboard. The page includes a form that allows users to input detailed information about unidentified bodies, including case numbers, descriptions, and other relevant data.

**Why this?**
This is essential for improving record-keeping and the search and trace process of unidentified individuals. It creates a central place for logging unidentified bodies and helps link cases to missing persons databases, streamlining communication between law enforcement and mortuary services.

**How are you doing this?**
Created a new Unidentified Body Page in the dashboard, accessible through the sidebar navigation.
Developed a form to capture critical data about unidentified bodies (case number, date of discovery, physical description, location, etc.).
Integrated form submission with the existing backend API to store data in the database.
Added validation to ensure required fields are completed before submission.
Designed the UI to be consistent with the rest of the dashboard, ensuring a user-friendly experience.

**List of Features:**
Unidentified Body Form: Capture case number, physical description (height, weight, scars, etc.), date, location of discovery, and other important data.
Form Validation: Ensures all required fields are completed before the data is submitted.
Backend Integration: Submits form data to the API and confirms successful entry into the database.
Responsive Design: Ensures that the form works on various screen sizes, including mobile devices.

**Type of change**
 New feature (non-breaking change which adds functionality)
How can it be tested?
Navigate to the Unidentified Body Page: Go to the sidebar menu and select the newly added "Unidentified Bodies" option to access the form.
Fill out the form: Input case information, physical description, and location details. Ensure that validation triggers if any required fields are left blank.
Submit the form: Verify that the data is successfully submitted and that you receive confirmation of entry into the system.
Test responsiveness: Ensure the form displays correctly on different devices (mobile, tablet, desktop).
API validation: Confirm that the submitted data is properly stored in the backend by reviewing the API responses.

**Screenshots**
![Screenshot from 2024-09-22 09-38-39](https://github.com/user-attachments/assets/d94afb96-e668-477b-a1a9-6269f2e15f6b)
![Screenshot from 2024-09-22 09-45-17](https://github.com/user-attachments/assets/4ac5449a-1ea1-49fd-96d1-9a1205c2e347)
![Screenshot from 2024-09-22 09-45-49](https://github.com/user-attachments/assets/3cab853e-0878-4035-b8d5-0c0022b677b7)

**JIRA ticket**
https://wambuinatalygakenye.atlassian.net/browse/UP-85?atlOrigin=eyJpIjoiNWU1NDJiNDdjMmI4NDMwNTg2OWI4OTdmODE1NDI5ZDUiLCJwIjoiaiJ9
